### PR TITLE
[KOGITO-9614] Fixing issue with async api and dev mode

### PR DESCRIPTION
**Pull request by fjtirado**
Reverted to previos version of async api base on code generation (since the alternative approach mengled with quarkus config management and make wrong assumption about resource location) 
There is a degree of classpath manipulation that I cannot avoid but it should be fine since it only occurs while on devmode. Also there are some exception while cleanup test that I feel are acceptable since it does not affect the overal result
```
2023-08-08 11:49:05,475 ERROR [io.qua.sma.rea.run.dev.ReactiveMessagingHotReplacementSetup] (pool-39-thread-1) Failed to redeploy application on changes: java.lang.IllegalStateException: Cannot reset a started application
	at io.quarkus.dev.appstate.ApplicationStateNotification.reset(ApplicationStateNotification.java:20)
	at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:90)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:222)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:203)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:537)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:437)
	at io.quarkus.smallrye.reactivemessaging.runtime.devmode.ReactiveMessagingHotReplacementSetup$OnMessage$1.run(ReactiveMessagingHotReplacementSetup.java:44)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### CODE REVIEW

**Comment by radtriste** [comment]
jenkins run kogito-runtimes quarkus-lts

**Comment by radtriste** [comment]
jenkins run kogito-runtimes native

**Comment by fjtirado** [comment]
Jenkins run kogito-runtimes quarkus-lts

**Comment by fjtirado** [comment]
Jenkins run kogito-runtimes native

**Comment by fjtirado** [comment]
Jenkins run kogito-runtimes quarkus-lts

**Comment by fjtirado** [comment]
Jenkins run kogito-runtimes native

File: quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-deployment/src/main/java/org/kie/kogito/quarkus/serverless/workflow/asyncapi/AsyncAPIProcessor.java [hunk 1]
```diff
@@ -15,21 +15,23 @@
...
+import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 
 public class AsyncAPIProcessor {
 
     @BuildStep
-    KogitoBuildContextAttributeBuildItem asyncAPIContext(List<AsyncAPIBuildItem> asyncAPIBuildItem) {
-        return new KogitoBuildContextAttributeBuildItem(ParserContext.ASYNC_CONVERTER_KEY, new AsyncAPIInfoConverter(
-                new MapAsyncAPIRegistry(asyncAPIBuildItem.stream().map(AsyncAPIBuildItem::getAsyncAPI).collect(Collectors.toList()))));
+    void asyncAPIContext(LiveReloadExecutionBuildItem reload, KogitoBuildContextBuildItem context, BuildProducer<KogitoAddonsPreGeneratedSourcesBuildItem> sources) throws ClassNotFoundException {
```
Comment (fjtirado) [inline quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-deployment/src/main/java/org/kie/kogito/quarkus/serverless/workflow/asyncapi/AsyncAPIProcessor.java:33]: LiveReload depends on KogitoBuildContext (which depends on KogitoContextAttributeBuildItem), so here the trick is to declare a producer (that actually does not produce anything) so this is invoked before kogito codegen and take benefit of the fact that KogitoBuildContext is a mutable class. 
We need LiveReload to use the proper classloader when loading the service

**Comment by kie-ci3** [comment]
**(quarkus-lts) - kogito-runtimes job** `#89` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>

export QUARKUS_BRANCH=2.13
export BUILD_MVN_OPTS_CURRENT=-T 1C 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-runtimes -u https://github.com/kiegroup/kogito-runtimes/pull/3165 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest.quarkus-lts/job/kogito-runtimes.quarkus-lts.kogito-runtimes/89/display/redirect

**Test results:**
- PASSED: 2206
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest.quarkus-lts/job/kogito-runtimes.quarkus-lts.kogito-runtimes/89/testReport/org.kie.kogito.quarkus.serverless.workflow.deployment.livereload/LiveReloadProcessorTest/testAsyncApi/">org.kie.kogito.quarkus.serverless.workflow.deployment.livereload.LiveReloadProcessorTest.testAsyncApi</a></summary>
1 expectation failed.<br/>Expected status code <201> but was <500>.<br/>
</details>

**Comment by kie-ci3** [comment]
**(quarkus-lts) - kogito-runtimes job** `#90` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>

export QUARKUS_BRANCH=2.13
export BUILD_MVN_OPTS_CURRENT=-T 1C 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-runtimes -u https://github.com/kiegroup/kogito-runtimes/pull/3165 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest.quarkus-lts/job/kogito-runtimes.quarkus-lts.kogito-runtimes/90/display/redirect

**Test results:**
- PASSED: 2206
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest.quarkus-lts/job/kogito-runtimes.quarkus-lts.kogito-runtimes/90/testReport/org.kie.kogito.quarkus.serverless.workflow.deployment.livereload/LiveReloadProcessorTest/testAsyncApi/">org.kie.kogito.quarkus.serverless.workflow.deployment.livereload.LiveReloadProcessorTest.testAsyncApi</a></summary>
1 expectation failed.<br/>Expected status code <201> but was <500>.<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#2966` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/kogito-runtimes/pull/3165 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/2966/display/redirect

**Test results:**
- PASSED: 2775
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/2966/testReport/org.kie.kogito.index/ProcessDataIndexOracleIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexOracleIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path data.UserTaskInstances[0].attachments.size() doesn't match.<br/>Expected: is <1><br/>  Actual: <0><br/>
</details>

**Comment by kie-ci3** [comment]
**(native) - kogito-runtimes job** `#51` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-Dnative -Dquarkus.native.container-build=true 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-runtimes -u https://github.com/kiegroup/kogito-runtimes/pull/3165 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest.native/job/kogito-runtimes.native.kogito-runtimes/51/display/redirect

**Test results:**
- PASSED: 3352
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest.native/job/kogito-runtimes.native.kogito-runtimes/51/testReport/org.kie.kogito.quarkus.workflows/AsyncAPIIT/testConsumer/">org.kie.kogito.quarkus.workflows.AsyncAPIIT.testConsumer</a></summary>
Condition with alias 'dead' didn't complete within 10 seconds because assertion condition defined as a lambda expression in org.kie.kogito.quarkus.workflows.AssuredTestUtils that uses java.lang.String, java.lang.Stringjava.lang.String 1 expectation failed.<br/>Expected status code <404> but was <200>.<br/>.
</details>

**Comment by fjtirado** [comment]
> @fjtirado https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest.quarkus-lts/job/kogito-runtimes.quarkus-lts.kogito-runtimes/90/testReport/org.kie.kogito.quarkus.serverless.workflow.deployment.livereload/LiveReloadProcessorTest/testAsyncApi/
> 
> Looks like the inheritance issue is still there ... is that related ?

I have my doubts, but now it is clear is not related with async api extension, but with the reload test.

File: quarkus/addons/messaging/common/src/main/java/org/kie/kogito/addon/quarkus/messaging/common/message/CloudEventHttpOutgoingDecorator.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 /**
  * Decorators for Http CloudEvents outgoing messages
  */
-public final class CloudEventHttpOutgoingDecorator implements MessageDecorator {
+public class CloudEventHttpOutgoingDecorator implements MessageDecorator {
```
Comment (fjtirado) [inline quarkus/addons/messaging/common/src/main/java/org/kie/kogito/addon/quarkus/messaging/common/message/CloudEventHttpOutgoingDecorator.java:28]: @radtriste This is intending to fix the inheritance issue on lts

**Comment by fjtirado** [comment]
Jenkins run kogito-runtimes quarkus-lts

**Comment by kie-ci3** [comment]
**(tests) - kogito-runtimes job** `#2992` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT=-T 1C -Dvalidate-formatting -Prun-code-coverage
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-runtimes -u https://github.com/kiegroup/kogito-runtimes/pull/3165 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/2992/display/redirect

**Test results:**
- PASSED: 3350
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/2992/testReport/org.kie.kogito.persistence.kafka/KafkaProcessInstancesIT/testFindByIdReadMode/">org.kie.kogito.persistence.kafka.KafkaProcessInstancesIT.testFindByIdReadMode</a></summary>
<br/>expected: 5<br/> but was: 0
</details>

**Comment by fjtirado** [comment]
jenkins run kogito-runtimes tests

**Pull request merged**